### PR TITLE
Add Facebook sign in test

### PR DESCRIPTION
**Pull request by mario-galic**
@jamespamplin @markjamesbutler 

[Example screencast](https://saucelabs.com/tests/c6869c26dc3240ad9693b75ce18be123)

This PR begins the migration of old social tests.

Facebook test credentials have been added to private `DEV/identity-frontend.conf`, and as environmental variables in TeamCity.

### CODE REVIEW

File: functional-tests/src/test/scala/util/Browser.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package test.util
+
+import org.openqa.selenium.support.ui.{ExpectedCondition, ExpectedConditions, WebDriverWait}
+import org.openqa.selenium.By
+import org.scalatest.selenium.WebBrowser
+import scala.util.Try
+
+trait Browser extends WebBrowser { this: WebBrowser =>
```
Comment (jamespamplin) [inline functional-tests/src/test/scala/util/Browser.scala]: Is `this: WebBrowser =>` needed here?

File: functional-tests/src/test/scala/util/Browser.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  def elementHasText(q: Query, text: String): Boolean = {
+    waitUntil(ExpectedConditions.textToBePresentInElementLocated(q.by, text))
+  }
+
+  def waitUntil[T](pred: ExpectedCondition[T]) = {
+    Try(new WebDriverWait(driver, timeOutSec).until(pred)).isSuccess
+  }
+
+  private val timeOutSec = 30
```
Comment (jamespamplin) [inline functional-tests/src/test/scala/util/Browser.scala]: Should be at the top of the trait for readability.

File: functional-tests/src/test/scala/util/Browser.scala (hunk 1 above)
Comment (mario-galic) [inline functional-tests/src/test/scala/util/Browser.scala]: Fixed.

File: functional-tests/src/test/scala/util/Browser.scala (hunk 2 above)
Comment (mario-galic) [inline functional-tests/src/test/scala/util/Browser.scala]: Fixed.

**Comment by jamespamplin** [comment]
:+1: great work

**Comment by mario-galic** [comment]
Passed in PROD: [Screencast](https://saucelabs.com/tests/053ccb92e34d45bd83c77c51e08e2c61)

**Pull request merged**